File: app/src/main/java/com/example/myapplication/codingminutes_levelup/binarysearch/BinarySearch.java
package com.example.myapplication.codingminutes_levelup.binarysearch;

public class BinarySearch {
    public static void main(String[] args) {
        int[] a = {2, 4, 6, 10, 15, 17, 22, 24, 30};
        int key = 24;

        int result = binarySearch(a, key);
        System.out.println("Result:" + result);
    }

    static int binarySearch(int[] arr, int key) {
        int s = 0;
        int e = arr.length - 1;

        while (s <= e) {
            int mid = (s + e) / 2;
            if (key == arr[mid]) {
                return mid;
            } else if (key > arr[mid]) {
                s = mid + 1;
            } else {
                e = mid - 1;
            }
        }

        return -1;
    }
}

File: app/src/main/java/com/example/myapplication/algorithm/searching/binarysearch/BinarySearch.kt
package com.example.myapplication.algorithm.searching.binarysearch

import kotlin.math.floor

fun main(){
    val array = intArrayOf(1,5,10,15,34,56,65,60,78,90)
    val result = binarySearch(array,1,0,array.size-1)
    println("RESULT::$result")

}

fun binarySearch(array: IntArray,target:Int,start:Int,end:Int):Int{

    val middle = floor(((start+end)/2).toDouble()).toInt()
    if(array[middle]==target){
        return middle
    }
    if(start==end){
        return -1
    }
    if(array[middle]<target){
        return binarySearch(array,target,middle+1,end)
    }
    if(array[middle]>target){
        return binarySearch(array,target,start,middle-1)
    }
    return -1
}